### PR TITLE
Spring migration

### PR DESCRIPTION
**Pull request by Rangel-Plachkov**

### CODE REVIEW

File: src/main/java/bg/fmi/javaweb/racemanagement/models/Event.java [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 @SuppressWarnings("UnusedDeclaration")
 public class Event {
-    private static final String DEFAULT_NAME = "Unknown models.Event";
+    private static final String DEFAULT_NAME = "Unknown Event";
```
Comment (Slavomir-Dreamix) [inline src/main/java/bg/fmi/javaweb/racemanagement/models/Event.java:8]: I am surprised by this constant,, especially it's value. Where do you need it? Sory for not checking myself, I cannot clone this project as collaborator (some privileges are missing).

File: src/main/java/bg/fmi/javaweb/racemanagement/models/Team.java [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 @SuppressWarnings("UnusedDeclaration")
 public class Team {
-    private static final String DEFAULT_NAME = "Unknown models.Team";
+    private static final String DEFAULT_NAME = "Unknown Team";
```
Comment (Slavomir-Dreamix) [inline src/main/java/bg/fmi/javaweb/racemanagement/models/Team.java:7]: I am wandering why is this value ever needed.

File: src/main/java/bg/fmi/javaweb/racemanagement/models/Track.java [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 @SuppressWarnings("UnusedDeclaration")
 public class Track {
-    private static final String DEFAULT_NAME = "Unknown models.Track";
+    private static final String DEFAULT_NAME = "Unknown Track";
```
Comment (Slavomir-Dreamix) [inline src/main/java/bg/fmi/javaweb/racemanagement/models/Track.java:5]: ...the same.

File: .gitignore [hunk 4]
```diff
@@ -1,30 +1,33 @@
-### IntelliJ IDEA ###
-out/
-!**/src/main/**/out/
-!**/src/test/**/out/
-.idea/
+HELP.md
```
Comment (Slavomir-Dreamix) [inline .gitignore:1]: This type of infrastructural changes (git ignore, the build files...) should be done in a separate commit. and name it accordingly. Also try to name commits what exactly they are doing. If you are not sure you can write "Attempt to migrate to a spring project and the maven build tool"

File: src/test/java/bg/fmi/javaweb/racemanagement3/RaceManagement3ApplicationTests.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package bg.fmi.javaweb.racemanagement3;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class RaceManagement3ApplicationTests {
+
+    @Test
+    void contextLoads() {
```
Comment (Slavomir-Dreamix) [inline src/test/java/bg/fmi/javaweb/racemanagement3/RaceManagement3ApplicationTests.java]: Well, so this test is doing almost nothing. It just check that the context can be loaded at all. Either delete it or do some meaningful unit testing.

File: src/main/java/bg/fmi/javaweb/racemanagement/repository/EventRepository.java [hunk 6]
```diff
@@ -11,6 +12,7 @@
 import java.util.stream.Collectors;
 
 @SuppressWarnings("UnusedDeclaration")
```
Comment (Slavomir-Dreamix) [inline src/main/java/bg/fmi/javaweb/racemanagement/repository/EventRepository.java:14]: I suggest not to use the @SuppressWarnings("UnusedDeclaration") let the warnings shine if not used, and either use it or delete ;). This warning is ok if the class is used as a dependency in other project for example. Or maybe for the controllers, who are called by the spring framework.

**Review by Slavomir-Dreamix — CHANGES_REQUESTED** [review]
Enjoy your projects.

File: src/test/java/bg/fmi/javaweb/racemanagement3/RaceManagement3ApplicationTests.java (hunk 5 above)
Comment (Rangel-Plachkov) [reply src/test/java/bg/fmi/javaweb/racemanagement3/RaceManagement3ApplicationTests.java]: As of right now, I want to leave it since this project is for learning purposes and it stays there as a  "reminder"
